src/outputs/restart_hdf5.cpp

```diff
@@ -134,6 +134,18 @@ RestartReaderHDF5::MeshInfo RestartReaderHDF5::GetMeshInfo() const {
   mesh_info.level_gid_lid_cnghost_gflag =
       ReadDataset<int>("/Blocks/loc.level-gid-lid-cnghost-gflag");
 
+  try {
+    mesh_info.derefinement_count = ReadDataset<int>("/Blocks/derefinement_count");
+  } catch (const std::runtime_error &e) {
+    // File does not contain this dataset, so must be older. Set to default value of zero
+    if (Globals::my_rank == 0)
```

Comment:
Extend to the check to only print with mesh refinement enabled?

Reply:
I think because of the HDF5 spew that comes out when the dataset doesn't exist, we should probably always leave the warning in to provide some explanation to the user of what is going on.

Reply:
Given that I removed the hdf5 error message, I now added the additional check.